### PR TITLE
Add test request logging

### PR DESCRIPTION
**Pull request by paulmey**
Provides some more context when integration tests fail

### CODE REVIEW

File: management/testutils/managementclient.go [hunk 1]
```diff
@@ -27,5 +27,62 @@ func GetTestClient(t *testing.T) management.Client {
 	if err != nil {
 		t.Fatal(err)
 	}
-	return client
+	return logger{client, t}
+}
+
+type logger struct {
```
Comment (ahmetalpbalkan) [inline management/testutils/managementclient.go]: `loggedClient` or `tracedClient` or `testClient` is better IMO

File: management/testutils/managementclient.go [hunk 2]
```diff
@@ -27,5 +27,62 @@ func GetTestClient(t *testing.T) management.Client {
...
+	management.Client
+	*testing.T
```
Comment (ahmetalpbalkan) [inline management/testutils/managementclient.go]: if you're not gonna let others do `client.Fatalf`, I think this should be privately embedded (i.e. `t *testing.T`).

File: management/testutils/managementclient.go [hunk 3]
```diff
@@ -27,5 +27,62 @@ func GetTestClient(t *testing.T) management.Client {
...
+}
+
+func (l logger) SendAzureDeleteRequest(url string) (management.OperationID, error) {
+	oid, err := l.Client.SendAzureDeleteRequest(url)
+	logOperation(l.T, "DELETE", url, nil, oid, err)
+	return oid, err
+}
+
+func logOperation(t *testing.T, method, url string, data []byte, oid management.OperationID, err error) {
+	t.Logf("AZURE> %s %s\n", method, url)
```
Comment (ahmetalpbalkan) [inline management/testutils/managementclient.go]: hmm time to use logrus (or something with fields) is coming perhaps...

File: management/testutils/managementclient.go [hunk 4]
```diff
@@ -27,5 +27,62 @@ func GetTestClient(t *testing.T) management.Client {
...
+
+	if len(s) > maxlen {
+		return s[:maxlen] + "..."
+	}
+	return s
+}
+
+func (l logger) SendAzureGetRequest(url string) ([]byte, error) {
+	d, err := l.Client.SendAzureGetRequest(url)
+	l.T.Logf("AZURE> GET %s\n", url)
```
Comment (ahmetalpbalkan) [inline management/testutils/managementclient.go]: rest of the method is duplicate with `logOperation`?

File: management/testutils/managementclient.go (hunk 2 above)
Comment (paulmey) [inline management/testutils/managementclient.go]: this func returns the `management.Client` interface, so T is not exposed?

File: management/testutils/managementclient.go (hunk 3 above)
Comment (paulmey) [inline management/testutils/managementclient.go]: baby steps. but yes.

File: management/testutils/managementclient.go (hunk 4 above)
Comment (paulmey) [inline management/testutils/managementclient.go]: I'll factor out the common parts

File: management/testutils/managementclient.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline management/testutils/managementclient.go]: ah true.

File: management/testutils/managementclient.go (hunk 1 above)
Comment (paulmey) [inline management/testutils/managementclient.go]: :+1: I'll go for testClient

**Comment by paulmey** [comment]
updated...

**Pull request merged**